pages/folder.tsx

```diff
@@ -0,0 +1,232 @@
+import React, { ChangeEvent, useEffect, useState } from 'react';
+import { useRouter } from 'next/router';
+import { getFolderInfo, getLinksInfo, getSelectLinksInfo } from './api/api';
+import Card from '@/components/Card';
+import styles from '@/styles/Folder.module.css';
+import { ModalData } from '@/components/ModalData';
+import close from '@/public/close.svg';
+
+interface FolderInfo {
+  id: number;
+  name: string;
+}
+
+interface LinksInfo {
+  createdAt: Date | undefined;
+  created_at: Date | undefined;
+  description: string | null;
+  imageSource: string | undefined;
+  image_source: string | null;
+  url: string | null;
+}
+
+type ModalOpenHandler = (name: string) => void;
+type ModalCloseHandler = () => void;
+
+export default function Folder() {
+  const [folderInfo, setFolderInfo] = useState<FolderInfo[]>([]);
+  const [linksInfo, setLinksInfo] = useState<LinksInfo[]>([]);
+  const [selectFolder, setSelectFolder] = useState<number | null>(null);
+  const [selectFolderName, setSelectFolderName] = useState<string>('전체');
+  const [isOpenModal, setIsOpenModal] = useState<boolean>(false);
+  const [selectModal, setSelectModal] = useState<any>(null);
+  const [search, setSearch] = useState<string>('');
+  const [filterLinks, setFilterLinks] = useState<LinksInfo[]>([]);
+  const router = useRouter();
+
+  const handleLoadFolderInfo = async () => {
+    try {
+      const data = await getFolderInfo();
+      const folderInfo = data.data;
+      const allFolderInfo = [{ id: 1, name: '전체' }, ...folderInfo];
+
+      if (!folderInfo) return;
+      setFolderInfo(allFolderInfo);
+    } catch (error) {
+      console.error(error);
+    }
+  };
```

Comment:
```suggestion
const handleLoadFolderInfo = async () => {
  try {
    const data = await getFolderInfo();
    if (!data || !data.data) {
      return; // 데이터가 유효하지 않으면 함수를 종료합니다.
    }

    const folderInfo = data.data;
    const allFolderInfo = [{ id: 1, name: '전체' }, ...folderInfo];
    setFolderInfo(allFolderInfo);
  } catch (error) {
    console.error(error);
  }
};
```

아래처럼 유효성 검사를 위로 올려야하지 않을까요!?